### ISSUE TITLE
"Something went wrong. Please try again." when trying to create conversation

### ISSUE THREAD

**Issue by MickMelon** [twilio/twilio-conversations-demo-react#37]
I followed the readme to setup this demo app using the Twilio function. I did not use Firebase.

I am able to login successfully, however when I try to create a conversation, a toast appears saying "Something went wrong. Please try again." 

Had a wee look to see if I can find out what's going wrong, and found that we are getting the following error in the `addConversation` method in `api.ts`

```
TypeError: Cannot read properties of undefined (reading 'then')
    at _class._callee8$ (client.ts:1070:1)
    at tryCatch (regeneratorRuntime.js:86:1)
    at Generator._invoke (regeneratorRuntime.js:66:1)
    at Generator.next (regeneratorRuntime.js:117:1)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at _next (asyncToGenerator.js:25:1)
```

This seems to be thrown by the following call: 

```ts
const conversation = await client.createConversation({
  friendlyName: name,
});
```

https://gyazo.com/dff0fb9b4a41a3768ce6d0c0d5b96f5c

Perhaps I'm missing something obvious here?

**Comment by jkabat** [comment]
I'm having the same problem. Also I do not see logged in users in Twilio console under registered identities even though access tokens are fine. Maybe some permission problem?

createConversation is failing for me, getSubscribedConversations as well...

```javascript
client.ts:1050 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'then')
    at _class._callee7$ (client.ts:1050:1)
    at tryCatch (regeneratorRuntime.js:86:1)
    at Generator._invoke (regeneratorRuntime.js:66:1)
    at Generator.next (regeneratorRuntime.js:117:1)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at _next (asyncToGenerator.js:25:1)
```

which comes from 

```javascript
  public async getSubscribedConversations(): Promise<Paginator<Conversation>> {
    await this._ensureReady;
    return this._conversationsPromise.then((conversations) =>
      conversations.getConversations()
    );
  }
```

**Comment by berkus** [comment]
@InfiniteRain pls take a look.

**Comment by InfiniteRain** [comment]
I will check this out ASAP and report back

**Comment by InfiniteRain** [comment]
I tried reinstalling the demoapp using a new token generator function, and everything seems to work fine for me. My suspicion is that there might be some permission defaults. I will investigate further.

**Comment by jkabat** [comment]
I have created new conversations service for demoapp and it seems to me that the permissions are the same as for the default conversations service.

**Comment by MickMelon** [comment]
Hey. I followed the readme again and it's now working as expected.

Last time, when I got to step 6 and it asked:

> Also navigate to Push configuration and enable all types of notifications for receiving FCM messages

I was using the Dark Reader Chrome extension, and it appeared to me that these checkboxes for these notifications were all enabled. Turns out, they were actually disabled. I'm not sure if this caused the problem though, as I've just tried disabling these again and it still works fine (i.e. I can create conversations, add participants, and communicate with other participants all as expected). 

Another difference was that this time I gave `user00` and `user01` different passwords. Don't believe this would cause such an issue the first time round though.

So, I'm not sure if it was just the process of creating a new token generator function that fixed it. Perhaps @InfiniteRain's suspicion about permission defaults is correct?

PS: This was using the same version of this demo as when I first created this issue (on commit ffc60f193b301b44fe7859ab926c4291b0c621e4)
